docs/ref/models/database-functions.txt

```diff
@@ -245,29 +245,32 @@ Usage example::
 .. versionadded:: 2.0
 
 Returns a positive integer corresponding to the 1-indexed position of the
-first occurrence of ``substring`` inside another string.
-If the substring is not found, returns 0.
+first occurrence of ``substring`` inside another string, or 0 if the substring
+is not found.
 
 Usage example::
 
     >>> from django.db.models.functions import StrIndex
     >>> Author.objects.create(name='Margaret Smith')
     >>> Author.objects.create(name='Smith, Margaret')
     >>> Author.objects.create(name='Margaret Jackson')
-    >>> authors = Author.objects.annotate(smith_index=StrIndex('name', 'Smith')).order_by('smith_index')
-    >>> print(articles.first().smith_at)
-    1
-    >>> authors = Author.objects.annotate(smith_index=StrIndex('Smith')).filter(smith_index__gt=0)
+    >>> authors = Author.objects.annotate(
+    ...     smith_index=StrIndex('name', 'Smith')).order_by('smith_index')
+    >>> authors.first().smith_index
+    0
+    >>> authors = Author.objects.annotate(
+    ...    smith_index=StrIndex('name', 'Smith')).filter(smith_index__gt=0)
     <QuerySet [<Author: Margaret Smith>, <Author: Smith, Margaret>]>
 
 .. warning::
 
-    Using ``StrIndex`` with MySQL  will return case-insensitive results.
+    MySQL's ``INSTR`` function used by ``StrIndex`` does a case-insensitive
```

Comment:
Yup. I think maybe we could change this note to be a small one pointing to that one? That way if anyone comes across it, they realize it's a fundamental thing, not just a quirk of `StrIndex`